File: generated-code/example/java/TcpReadWrite/src/main/java/Runner.java
import java.io.*;
import java.net.Socket;

import util.StreamUtil;

public class Runner {
    public static void main(String[] args) throws IOException {
        if (args.length != 2) {
            throw new RuntimeException("Pass host and port as parameters");
        }
        String host = args[0];
        int port = Integer.parseInt(args[1]);

        Socket socket = new Socket(host, port);
        socket.setTcpNoDelay(true);

        InputStream inputStream = new BufferedInputStream(socket.getInputStream());
        model.Example input = model.Example.readFrom(inputStream);

        System.out.println(input);

        OutputStream outputStream = new BufferedOutputStream(socket.getOutputStream());
        input.writeTo(outputStream);
        outputStream.flush();

        socket.close();
    }
}

File: generated-code/csharp/example/TcpReadWrite/Runner.cs
using System;
using System.IO;
using System.Net.Sockets;

namespace TransGenTest
{
    public class Runner
    {
        public static void Main(string[] args)
        {
            string host = args[0];
            int port = int.Parse(args[1]);
            bool stdout = bool.Parse(args[2]);

            using (var tcpClient = new TcpClient(host, port))
            {
                using (var stream = new BufferedStream(tcpClient.GetStream()))
                {
                    var reader = new BinaryReader(stream);
                    var writer = new BinaryWriter(stream);
                    while (reader.ReadBoolean())
                    {
                        Example input = Example.ReadFrom(reader);
                        if (stdout)
                        {
                            Console.WriteLine(input.ToString());
                        }
                        input.WriteTo(writer);
                        writer.Flush();
                    }
                }
            }
        }
    }
}